orbit/cmd/orbit/orbit.go

```diff
@@ -516,6 +518,36 @@ func main() {
 		}
 		g.Add(flagRunner.Execute, flagRunner.Interrupt)
 
+		// only setup extensions autoupdate if we have enabled updates
+		if !c.Bool("disable-updates") || c.Bool("dev-mode") {
+			const orbitExtensionUpdateInterval = 60 * time.Second
+			extRunner := update.NewExtensionConfigUpdateRunner(orbitClient, update.ExtensionUpdateOptions{
+				CheckInterval: orbitExtensionUpdateInterval,
+				RootDir:       c.String("root-dir"),
+			}, updateRunner)
+
+			if _, err := extRunner.DoExtensionConfigUpdate(); err != nil {
+				// just log, OK to continue since this will get retry
+				log.Info().Err(err).Msg("initial update to fetch extensions from /config API failed")
+			}
+
+			// call UpdateAction on the updateRunner after we have fetched extensions from Fleet
+			_, err := updateRunner.UpdateAction()
```

Comment:
Am worried about what would happen if any of the non-extension targets were updated (orbit, osquery, desktop), the way it looks it would just update but continue running the old version (we'd miss the update).

Maybe this is low probability bug? (We already call `updateRunner.UpdateAction()` and restart accordingly in line 301.)

Reply:
I wouldn't quite categorize this as a bug..? The call to `UpdateAction()` on line `301` happens at every `update-interval`, and I specifically rely on it, and since it's added to the oklog run group, it does restart things correctly. 

I am using the "extra" `UpdateAction()` call here to kick start things after orbit is enrolled and extensions are fetched from agent options.

Reply:
Sorry, am not following:
1. Line 534 calls `UpdateAction`; say it does update Orbit, osquery or desktop. First return value is ignored.
2. Orbit continues to run as usual.
3. UpdateAction is executed every update-interval but it will not detect the new update, because it already took place.

It's an edge case for sure.

Reply:
Ah, I see. Yeah this is an edge case. I do want to have this initial call to `UpdateAction` so that extensions can be quickly loaded. I think perhaps, I can break this function in two, and handle the case where only extension related `UpdateAction` is called, but I am thinking that this could be handled in a follow up, as a part of #9046?